rovi2/Q q
rovi2/Q dq
bool is_moving
bool is_moving2
bool e
bool valid
